File: Examples/test-suite/ocaml/special_variable_macros_runme.ml
open Swig
open Special_variable_macros

let _ =
  let name = new_Name '() in
  assert (_testFred '(name) as string = "none");
  assert (_testJack '(name) as string = "$specialname");
  assert (_testJill '(name) as string = "jilly");
  assert (_testMary '(name) as string = "SWIGTYPE_p_NameWrap");
  assert (_testJames '(name) as string = "SWIGTYPE_Name");
  assert (_testJim '(name) as string = "multiname num");
  let arg = new_PairIntBool '(10, false) in
  assert (_testJohn '(arg) as int = 123);
  assert (_makeStringInt '("stringint", 999) as string = "stringint");
  assert (_provideStringInt '(999) as string = "1000");
  assert (_shortFunction '(1, 1) as int = (200*2 + 200*3));
;;

File: Examples/test-suite/go/special_variable_macros_runme.go
package main

import "swigtests/special_variable_macros"

func main() {
	name := special_variable_macros.NewName()
	if special_variable_macros.TestFred(name) != "none" {
		panic("test failed")
	}
	if special_variable_macros.TestJack(name) != "$specialname" {
		panic("test failed")
	}
	if special_variable_macros.TestJill(name) != "jilly" {
		panic("test failed")
	}
	if special_variable_macros.TestMary(name) != "SWIGTYPE_p_NameWrap" {
		panic("test failed")
	}
	if special_variable_macros.TestJames(name) != "SWIGTYPE_Name" {
		panic("test failed")
	}
	if special_variable_macros.TestJim(name) != "multiname num" {
		panic("test failed")
	}
	if special_variable_macros.TestJohn(special_variable_macros.NewPairIntBool(10, false)) != 123 {
		panic("test failed")
	}
	if special_variable_macros.MakeStringInt("stringint", 999) != "stringint" {
		panic("test failed")
	}
	if special_variable_macros.ProvideStringInt(999) != "1000" {
		panic("test failed")
	}
	if special_variable_macros.ShortFunction(1, 1) != (200*2 + 200*3) {
		panic("test failed");
	}
}
